src/main/kotlin/notary/endpoint/eth/EthRefundStrategyImpl.kt

```diff
@@ -27,21 +27,26 @@ class NotaryException(reason: String) : Exception(reason)
  * Class performs effective implementation of refund strategy for Ethereum
  */
 class EthRefundStrategyImpl(
-    private val notaryConfig: EthNotaryConfig,
+    notaryConfig: EthNotaryConfig,
     private val irohaNetwork: IrohaNetwork,
     private val credential: IrohaCredential,
     ethereumConfig: EthereumConfig,
     ethereumPasswords: EthereumPasswords,
     private val tokensProvider: EthTokensProvider
 ) : EthRefundStrategy {
 
-    val relayProvider = EthRelayProviderIrohaImpl(
+    private val relayProvider = EthRelayProviderIrohaImpl(
         irohaNetwork,
         credential,
         credential.accountId,
         notaryConfig.registrationServiceIrohaAccount
     )
 
+    private val whiteListProvider = WhiteListProvider(
```

Comment:
What about abstract class whiteListProvider and two descendants: EthWhitelist and BtcWhitelist? Maybe this way it will be a better design

Reply:
This is a quite good idea. Yes, I will do this.